### DISCUSSION TITLE
Question about the runtime error: nodata value within domain while Reading LAI

### DISCUSSION THREAD

**Discussion by LuvMangorange** [mhm-ufz/mHM#23] [Q&A] [answered]
Hi!
There is a problem occurred during runtime about 'get time variable'.
I have tried to debug it, and found the place, but failed to solve it. Besides the format of time variable is integer.
There are some pictures about the error and the lai.zip containing part of lai data. 
Thanks for help!

Note: Please do not focus on the real value of lai, the fillvalue has not been processed yet.

![image](https://user-images.githubusercontent.com/34867664/122681805-b7da6480-d228-11eb-9ae6-9a6e92ab2a52.png)
![image](https://user-images.githubusercontent.com/34867664/122718574-b0a26d80-d29f-11eb-9c92-64272108ffc8.png)
![image](https://user-images.githubusercontent.com/34867664/122681994-99289d80-d229-11eb-91e5-ecfd6ebc61cb.png)
[lai.zip](https://github.com/mhm-ufz/mhm/files/6682736/lai.zip)

**Comment by ViniciusBMoreira** [comment]
@LuvMangorange 

As long as the developers don't respond, I'll try to help you. Try changing the time reference of your lai.nc file (you can check it with the command "cdo sinfov lai.nc") with this command:

cdo setreftime,DATE1,00:00:00,days lai.nc lai_FINAL.nc

Your file have RefTime = 2001-01-01:

![image](https://user-images.githubusercontent.com/61054098/122694383-86a36980-d213-11eb-801c-0d58455af792.png)

Also look at the temporal resolution of the LAI data in the mhm configuration file !

Cheers
Bryan

**Reply by LuvMangorange** [reply]
@BryanSouzaM 
Thanks for your kind help!

As your advice, I have changed the reftime of lai to 2014-1-1, but it didn`t work.
However, from your response, I guess the problem may be caused by the projection, because the projection of my sample is epsg:2381(Xian_1980_3_Degree_GK_CM_108E).

Anyway, thank you!

**Comment by pallavhydro** [comment, marked as answer]
Hi @LuvMangorange 

My comments -

1. If we simply look at the runtime error, it says `nodata value within domain` while *Reading LAI*. That means there are `nodata` in LAI within the domain/ watershed. The domain boundary in mhm is set by  the `dem`. By the look of your lai missing values (almost half of your lai.nc is missing) since WeiHe watershed doesn't have a coast, I am assuming you are clipping lai.nc to your domain boundary. My suggestion would be to leave all mhm input (including lai.nc) unclipped i.e. full values over the rectangle extent, except for the dem. Then mhm will take care of correctly clipping out all data internally.
2. In order to check above point, you can try plotting lai.nc and dem in GIS. There should be one or more pixels where the domain extent/ boundary of dem misses lai value (share the screenshots with us if possible).
3. Request - I am not sure if you can change it now, but it would've been better if the title of the discussion is based on the runtime error, in this case  `nodata value within domain while Reading LAI`. Such title would help others to find this discussion who come across this issue in future.   

Let us know if you could move forward.

Cheers
Pallav

**Reply by LuvMangorange** [reply]
Hi! @pallavhydro 
The error has been solved mainly follow your tips.
But there are some differences, as my data was all made on GEE which has been cliped before downloaded. So I converted NoData to Number, and then the error disappered.
Thank you again!
